### ISSUE TITLE
Problematic SPSA Initialization

### ISSUE THREAD

**Issue by coatless** [mlpack/ensmallen#74]
With ensmallen 1.13.0 adding `SPSA`, there was a regression introduced. In particular, the `SPSA` initialization of parameters is set to some unknown value. This was picked up under the `-Wreorder` flag.

Details: https://stackoverflow.com/questions/1828037/whats-the-point-of-g-wreorder

Error Information:

```bash
* checking whether package 'RcppEnsmallen' can be installed ... WARNING
Found the following significant warnings:
  ../inst/include/ensmallen_bits/spsa/spsa.hpp:133:10: warning: 'ens::SPSA::Ak' will be initialized after [-Wreorder]
  ../inst/include/ensmallen_bits/spsa/spsa.hpp:123:10: warning:   'size_t ens::SPSA::maxIterations' [-Wreorder]
  ../inst/include/ensmallen_bits/spsa/spsa_impl.hpp:24:8: warning:   when initialized here [-Wreorder]
See 'd:/RCompile/CRANguest/R-release/RcppEnsmallen.Rcheck/00install.out' for details.
```

/ cc coatless/rcppensmallen#8

**Comment by zoq** [comment]
Thanks for the notification, https://github.com/mlpack/ensmallen/pull/75 will fix the issue.

**Comment by coatless** [comment]
@zoq that was quick! I almost had the PR ready.

**Comment by zoq** [comment]
Will make sure to wait a couple of minutes before opening the PR :) Also, will make sure the build process marks similar issues in the future.

**Comment by rcurtin** [comment]
Sorry for the slowness on this one.  I merged #75 and released 1.13.1.  Thanks for the report and feel free to open more issues for stuff like this in the future---it's pretty easy to fix. :)
